src/proxy/hdrs/XPACK.cc

```diff
@@ -245,12 +245,12 @@ XpackDynamicTable::lookup(uint32_t index, const char **name, size_t *name_len, c
     return {0, XpackLookupResult::MatchType::NONE};
   }
 
-  if (index < this->_entries[(this->_entries_tail + 1) % this->_max_entries].index) {
+  if (index < this->_entries[this->_calc_index(this->_entries_tail, 1)].index) {
     // The index is invalid
     return {0, XpackLookupResult::MatchType::NONE};
   }
 
-  uint32_t pos = (this->_entries_head - (this->_entries[this->_entries_head].index - index)) % this->_max_entries;
+  uint32_t pos = this->_calc_index(this->_entries_head, -(this->_entries[this->_entries_head].index - index));
```

Comment:
After thinking through this a bit, I see that this is a formulaic conversion from the previous version to this new `_calc_index` version. So it looks correct. That said, I suggest we go a step beyond and make the `offset` argument easier to read via:

```
index - this->_entries[this->_entries_head].index 
```

Reply:
Good suggestion. Thanks.